### PR TITLE
Adds a very low chance for items to get stuck in vending machines

### PR DESCRIPTION
**Pull request by Gamer025**
## About The Pull Request
This PR adds a small chance for items to get stuck in vending machines.
Items can be freed by buying a second item of that type or by bashing the vending machine.
This PR decreased crush and freebie chance and added a massive chance to unstuck a random stuck item.

Also adds tilt call to throw_impact on request from Ryll

## Why It's Good For The Game
Items getting stuck adds a neat little detail that makes vending machines a little more interesting
Since you can free the item with a very high chance by attacking the vendor the impact to gameplay should be very little (especially since the chance of an item getting stuck is already quite low)
This should also lead to more interactions with vending machines, like someone trying to free his item and getting crushed not knowing that it even possible (getting crushed also frees the stuck items)

## Changelog
:cl:
add: Items have now a very low chance to get stuck in vending machines
admin: Throwing vending machines at people now crushes them, like they attacked the vending machine
/:cl:

### CODE REVIEW

**Comment by Jared-Fogle** [comment]
time for people to have more reason to hate economy because they will buy items and not get them

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -1194,3 +1195,48 @@ GLOBAL_LIST_EMPTY(vending_products)
...
+ * * product - The item that should be vended
+ */
+/obj/machinery/vending/proc/vend_item(var/datum/data/vending_product/product)
+	var/obj/item/vended_item = new product.product_path()
+	product.amount--
+	if (prob(3))
+		//Item got stuck in vending machine
+		LAZYADD(stuck_items, vended_item)
+		to_chat(usr, "<span class='warning'>It seems like [product.name] got stuck in the vending machine!<br>\
+		Maybe you can get it unstuck by buying another [product.name] or by bashing the vending machine.</span>")
```
Comment (PKPenguin321) [inline code/modules/vending/_vending.dm:1211]: This is a bit too much, you would think they could figure out how to get it unstuck by themselves. Just "[product.name] gets stuck in the vending machine!" would suffice

Comment (Gamer025) [reply code/modules/vending/_vending.dm:1211]: I guess people could figure out to just bash the vending machine, but im not sure if they would realize that ordering the same item can also get it unstuck? Maybe they could think all types of that item are now stuck and not order it anymore?
I also want to really make sure that people know they can just punch the vendor and get the item very likely back.

**Comment by Jared-Fogle** [comment]
image

this is my design doc

**Comment by Timberpoes** [comment]
It's Christmas. It's the time of the year for cheer, for merriment, for goodwill between all men, women and other.

Also, closed. Vending machines are just a utility we have that let players exchange currency for items. They should be nothing more. And they are nothing less.

**Comment by Iamgoofball** [comment]
24 hour rule violation in my 2020? it's more likely than you think

**Pull request closed without merging**